packages/bpk-component-autosuggest/src/BpkAutosuggestV2/BpkAutosuggest.tsx

```diff
@@ -0,0 +1,596 @@
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    useEffect(() => {
+      if (highlightedIndex === previousHighlightedIndexRef.current) return;
+
+      previousHighlightedIndexRef.current = highlightedIndex;
+
+      const currentSuggestion =
+        highlightedIndex != null && highlightedIndex >= 0
+          ? (flattenedSuggestions?.[highlightedIndex] ?? null)
+          : null;
+
+      onSuggestionHighlighted?.({ suggestion: currentSuggestion });
+    }, [highlightedIndex, flattenedSuggestions, onSuggestionHighlighted]);
+
+    const handleInputInteraction = () => {
+      hasInteractedRef.current = true;
+      if (shouldRenderSuggestions) {
+        shouldRenderSuggestions(inputValue);
+        openMenu();
+      }
+
+      if (!isOpen && inputValue.length) {
+        onSuggestionsFetchRequested(inputValue);
+        openMenu();
+      } else {
+        onClick?.();
+      }
+
+      // Desktop destination autosuggest lives on the homepage and is "loaded/interacted with" via clicking on it
+      // Every other use case is within a new screen or modal so is interacted with via the user navigating into the modal/new screen
+      if (isDesktop) {
+        onLoad?.(inputValue);
+      }
+    };
+
+    const onKeyDown = (e: KeyboardEvent): void => {
+      if (e.key !== 'Enter') {
+        return;
+      }
+
+      if (e.key === 'Enter' && suggestions.length) {
+        onSuggestionSelected?.({ suggestion: suggestions[0], inputValue });
+      }
+
+      if (defaultValue) {
+        handleInputInteraction();
+      } else if (suggestions.length === 0) {
+        onSuggestionSelected?.();
+      }
+    };
+
+    const handleSuggestionClick = () => {
+      if (!focusInputOnSuggestionClick) {
+        (document.activeElement as HTMLElement)?.blur?.();
+      }
+    };
+
+    const clearSuggestions = (e?: SyntheticEvent<HTMLButtonElement>) => {
+      e?.stopPropagation();
+      setInputValue('');
+    };
+
+    // Render suggestions function to render single section suggestion
+    const renderSuggestions = (items: any[], sectionIndex?: number) =>
+      items.map((suggestion, index) => {
+        const suggestionIndex = sectionIndex ? index + sectionIndex : index;
+        const isFirstSuggestion = sectionIndex
+          ? sectionIndex === 0 && index === 0
+          : index === 0;
+
+        const key = getSuggestionValue(suggestion);
+
+        return (
+          <li
+            key={key}
+            {...getItemProps({
+              item: suggestion,
+              index: suggestionIndex,
+              onClick: handleSuggestionClick,
+            })}
+            className={getClassName(
+              theme.suggestion,
+              highlightedIndex === suggestionIndex &&
+                theme.suggestionHighlighted,
+              highlightFirstSuggestion &&
+                isFirstSuggestion &&
+                highlightedIndex === -1 &&
+                theme.suggestionHighlighted,
```

Comment:
would it be more readable?
```suggestion
          [theme.suggestionHighlighted]: highlightedIndex === suggestionIndex || (highlightFirstSuggestion && isFirstSuggestion && highlightedIndex === -1)
```